### ISSUE TITLE
[openimageio] build failure

### ISSUE THREAD

**Issue by etheory** [microsoft/vcpkg#10542]
**Host Environment**
- OS: Microsoft Windows 10 Home, Version 10.0.18362 Build 18362
- Compiler: VS2019 v16.5.0

**To Reproduce**
`./vcpkg install openimageio:x64-windows`

**Failure logs** 

This gets run directly from the "x64 Native Tools Command Prompt for VS 2019".
Everything proceeds fine until boost-mp11, then it goes pear-shaped:

```
Building package boost-mp11[core]:x64-windows... done
Installing package boost-mp11[core]:x64-windows...
The following files are already installed in Z:/Luke/Programming/Graphics/vcpkg/installed/x64-windows and are in conflict with boost-mp11:x64-windows

Installed by boost-chrono:x64-windows
    include/boost/chrono.hpp
    include/boost/chrono/ceil.hpp
    include/boost/chrono/chrono.hpp
    include/boost/chrono/chrono_io.hpp
    include/boost/chrono/clock_string.hpp
    include/boost/chrono/config.hpp
    include/boost/chrono/detail/inlined/chrono.hpp
    include/boost/chrono/detail/inlined/mac/chrono.hpp
    include/boost/chrono/detail/inlined/mac/process_cpu_clocks.hpp
    include/boost/chrono/detail/inlined/mac/thread_clock.hpp
    include/boost/chrono/detail/inlined/posix/chrono.hpp
    include/boost/chrono/detail/inlined/posix/process_cpu_clocks.hpp
    include/boost/chrono/detail/inlined/posix/thread_clock.hpp
    include/boost/chrono/detail/inlined/process_cpu_clocks.hpp
    include/boost/chrono/detail/inlined/thread_clock.hpp
    include/boost/chrono/detail/inlined/win/chrono.hpp
    include/boost/chrono/detail/inlined/win/process_cpu_clocks.hpp
    include/boost/chrono/detail/inlined/win/thread_clock.hpp
    include/boost/chrono/detail/is_evenly_divisible_by.hpp
    include/boost/chrono/detail/no_warning/signed_unsigned_cmp.hpp
    include/boost/chrono/detail/scan_keyword.hpp
    include/boost/chrono/detail/static_assert.hpp
    include/boost/chrono/detail/system.hpp
    include/boost/chrono/duration.hpp
    include/boost/chrono/floor.hpp
    include/boost/chrono/include.hpp
    include/boost/chrono/io/duration_get.hpp
    include/boost/chrono/io/duration_io.hpp
    include/boost/chrono/io/duration_put.hpp
    include/boost/chrono/io/duration_style.hpp
    include/boost/chrono/io/duration_units.hpp
    include/boost/chrono/io/ios_base_state.hpp
    include/boost/chrono/io/time_point_get.hpp
    include/boost/chrono/io/time_point_io.hpp
    include/boost/chrono/io/time_point_put.hpp
    include/boost/chrono/io/time_point_units.hpp
    include/boost/chrono/io/timezone.hpp
    include/boost/chrono/io/utility/ios_base_state_ptr.hpp
    include/boost/chrono/io/utility/manip_base.hpp
    include/boost/chrono/io/utility/to_string.hpp
    include/boost/chrono/io_v1/chrono_io.hpp
    include/boost/chrono/process_cpu_clocks.hpp
    include/boost/chrono/round.hpp
    include/boost/chrono/system_clocks.hpp
    include/boost/chrono/thread_clock.hpp
    include/boost/chrono/time_point.hpp
    include/boost/chrono/typeof/boost/chrono/chrono.hpp
    include/boost/chrono/typeof/boost/ratio.hpp

Please ensure you're using the latest portfiles with `.\vcpkg update`, then
submit an issue at https://github.com/Microsoft/vcpkg/issues including:
  Package: boost-mp11:x64-windows
  Vcpkg version: 2020.02.04-nohash-external
```

**Additional context**
NA

**Comment by JackBoosY** [comment]
Hi @etheory, could you remove `boost-mp11:x64-windows` and try again using powershell?

Thanks.

**Comment by etheory** [comment]
No good, still does the same thing.

What I did exactly was this:

1.) .\vcpkg.exe remove boost-mp11:x64-windows
2.) load powershell, then run `C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Auxiliary\Build\vcvarsamd64_x86.bat`
3.) .\vcpkg.exe install openimageio:x64-windows

Same issue, same errors from log.

**Comment by JackBoosY** [comment]
@etheory I think you don't need to execute step 2, just run command `.\vcpkg.exe install openimageio:x64-windows` would be okay.

**Comment by etheory** [comment]
But if I do that, how do I control which toolchain it uses? You can't directly control which toolchain vcpkg compiles with.

Regardless it still doesn't work.

**Comment by etheory** [comment]
Updating to VS2019 v16.5.1 still has the same issue.

**Comment by JackBoosY** [comment]
See [vcpkg_platform_toolset](https://github.com/microsoft/vcpkg/blob/master/docs/users/triplets.md#vcpkg_platform_toolset), you can add `set(vcpkg_platform_toolset v142)` to the selected triplet file(_VCPKG_PATH/triplets_).

**Comment by etheory** [comment]
Thanks, for one last time I've tried completely removing vcpkg, starting from scratch and trying within powershell.

I'll let you know how it goes.

**Comment by etheory** [comment]
Now I get a different error:

Downloading git...
  https://github.com/git-for-windows/git/releases/download/v2.25.1.windows.1/PortableGit-2.25.1-32-bit.7z.exe -> Z:\Luke\Programming\Graphics\vcpkg\downloads\PortableGit-2.25.1-32-bit.7z.exe
Extracting git...
Failed to do post-extract rename-in-place.
fs.rename(Z:\Luke\Programming\Graphics\vcpkg\downloads\tools\git-2.25.1-1-windows.partial.17252, Z:\Luke\Programming\Graphics\vcpkg\downloads\tools\git-2.25.1-1-windows, operation not permitted)

So basically, vcpkg seems incredibly unreliable.

**Comment by JackBoosY** [comment]
@etheory That's a vcpkg bug(#9132), please rename _Z:\Luke\Programming\Graphics\vcpkg\downloads\tools\git-2.25.1-1-windows.partial.17252_ to _Z:\Luke\Programming\Graphics\vcpkg\downloads\tools\git-2.25.1-1-windows_ manually, then try again.

Thanks.

**Comment by etheory** [comment]
OK, working my way manually by helping it with the renaming errors now. Crossing my fingers it works this time. Still a way to go....

**Comment by etheory** [comment]
OK, after nearly a whole day of running the process over and over again, manually renaming everything as I went it FINALLY installed. Not a very pleasant experience, but it worked. Thanks.

**Comment by etheory** [comment]
Closing this as it's mostly due to: https://github.com/microsoft/vcpkg/issues/9132
